astropy/io/fits/tests/test_diff.py

```diff
@@ -767,7 +810,6 @@ def test_file_output_overwrite_vs_clobber(self):
         hb['C'] = 4
         diffobj = HeaderDiff(ha, hb)
         diffobj.report(fileobj=outpath)
-        report_as_string = diffobj.report()
```

Comment:
I guess this was done to check that with `overwrite` not warning is raised, though it's weird that `overwrite=True` is not specified, not sure. But it's better not to change unrelated things.

Reply:
`report_as_string` is defined and not used.  `flake8` caught it for me.  I can back out this change if desired.

Reply:
Actually yes it's probably useless since it does not even write the report to a file. Fine to remove.